starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,490 @@
+//for (let i = 0; i < arr.length; i++) {
+//  if ($number < $arr[i]) {
+///  maximumNumber = arr[i];
+///}
+//}
+//return maximumNumber;
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(arr) {
+  let word = "";
+  if (arr.length === 0) {
+    return null;
+  }
+  for (let i = 0; i < arr.length; i++) {
+    if (word.length < arr[i].length) {
+      word = arr[i];
+    }
+  }
+  return word;
+}
+
+//const length = words[i].length;
+//for (i = 0; i < length; i++) {
+//  console.log("that is not the longest word");
+//}
+
+//for (let i = 0; i < words.length; i += 1) {
+//  console.log(studentsCountries[i]);
+//}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+//numbers.forEach(callback(number))
+//function sum(numbers.forEach(callback)) {
+//return a + b;}
+
+function sumArray(placeholder) {
+  let sum = 0;
+
+  for (let i = 0; i < placeholder.length; i++) {
+    sum += placeholder[i];
+  }
+  return sum;
+}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(placeholder) {
+  let sum = 0;
+  if (placeholder.length === 0) {
+    return null;
+  }
+  for (let i = 0; i < placeholder.length; i++) {
+    sum += placeholder[i];
+  }
+  let average = sum / placeholder.length;
+  return average;
+}
+
 // Level 2: Array of strings
 const wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
+function averageWordLength(placeholder) {
+  let sum = 0;
+  if (placeholder.length === 0) {
+    return null;
+  }
+  for (let i = 0; i < placeholder.length; i++) {
+    sum += placeholder[i].length;
+  }
+  let average = sum / placeholder.length;
+  return average;
+}
+
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
 
+function uniquifyArray(placeholder) {
+  let uniqueWord = 0;
+  if (placeholder.length === 0) {
+    return [];
+  }
+  for (let i = 0; i === placeholder.length; i++) {
+    //const index;
+    let index = placeholder.indexOf(i);
+    if (placeholder.indexOf(i) > -1) {
+      placeholder.splice(index, 1);
+      uniquifyArray(placeholder); //should return  uniquified array but redundand
+    }
+
+    uniquifyArray(console.log(placeholder)); //should return  uniquified array
```

Comment:
Another small comment here, you are passing a console.log as a parameter in your fuction uniquify, when what you actually want is to pass is a the array you are about to uniquify.